src/objectid.ts

```diff
@@ -60,7 +60,7 @@ export class ObjectId {
       this.__id = id.__id;
     }
 
-    if (typeof id === 'object' && 'id' in id) {
+    if (typeof id === 'object' && id && 'id' in id) {
```

Comment:
why is this change needed? the `typeof` check should prove the existence of the object (it's not `typeof === undefined`)

Reply:
`typeof null` is 'object', so this is checking for that